_posts/2022-12-09-trino-summit-2022-shopify-recap.md

```diff
@@ -0,0 +1,77 @@
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-summit-2022/shopify.jpg
+---
+
+Rolling right along with another one of [our Trino Summit 2022 recap posts]({% post_url
+2022-11-21-trino-summit-2022-recap %}), we're excited to bring you the engaging
+talk from Marc Laforet at Shopify. He talked about the ordeal (or, if you look
+at it in a positive light, the privilege) of migrating petabytes of data from
+Hive to Iceberg with the help of Trino. With details on why Shopify chose to
+move to Iceberg, the various migration strategies that were considered, and the
+ultimate process of moving all that data while the Trino Icerberg connector was
+still in active development, it's an insightful talk that you don't want to
+miss.
+
+<!--more-->
+
+{% youtube nJBBw-xnLU8 %}
+
+## Recap
+
+As with our other talks, it should come as no surprise that Shopify has a lot of
+data to work with. First-party data comes in from a few different sources, and
+there's a mountain of modelled data to go along with it. And as we know, there
+are always data challenges in companies that scale with massive amounts of the
+stuff. In Shopify's case, the issue was that some data sets were built on top of
+custom data storage formats, and the architecture wasn't scaled with a careful
+plan in mind, leading to limited cross-compatibility. With data scientists
+unable to unify data across different tools and storages, it was time for a
+change.
+
+When you've got tons of data that isn't currenly in one place, what's the fix?
+Create a central lakehouse for all the data to be accessible from, a
+single-service portal that could serve all users' needs. The first question was
+which table format to use, and if the title of the blog post didn't already give
+it away, the easy choice was to go with Iceberg. It was an easy, central vision
+to work towards: all data in a centralized lakehouse stored in Iceberg, then
+queryable by Trino.
+
+Having a plan and putting that plan into action are two different things,
+though, and when nothing is already in Iceberg, moving it all there is a
+migration on the scale of thousands of tables and petabytes of data. In Marc's
+words from the talk, once Shopify committed to the migration and invested
+resources into it, the realization was, "crap, now I have to build it." Even
+worse, because the old data was primarily in a Hive JSON format, it all needed
+to be rewritten... and so it was.
+
+Then, enter Trino! With new Iceberg-based tables, Trino was identified as the
+right tool for the job to query all that data. This wasn't without snags, as the
+migration happened while the Iceberg connector was still being aggressively
+worked on and developed. There were a few different incidents where Shopify hit
+a snag or an issue, and an update or bugfix to Trino's Iceberg connector solved
+those problems in a matter of days or weeks.
+
+The result of all of this? Some incredible benchmarks. Large tables saw a 96%
```

Comment:
Would you mind explaining how you came up with these percents?

Reply:
I'm using the numbers from your benchmark slides. (prev - current) / prev for all three.